packages/flutter_test/lib/src/finders.dart

```diff
@@ -460,57 +468,521 @@ class CommonFinders {
+  @protected
+  SemanticsNode get root;
+
+  @override
+  Iterable<SemanticsNode> get allCandidates {
+    return collectAllSemanticsNodesFrom(root);
+  }
+}
+
+/// A base class for creating [SemanticsNode] [FinderBase]s.
+abstract class SemanticsFinder extends FinderBase<SemanticsNode> with SemanticsFinderMixin {
+  /// Creates a new [SemanticsFinder] that will search starting at the given
+  /// [root].
+  SemanticsFinder(this.root);
+
+  @override
+  final SemanticsNode root;
+}
+
+/// Class that wraps an implementation of [FinderBase<Element>] and implements
+/// [Finder] for backwards compatibility with the previous [Finder] API.
+class _FinderBox implements Finder {
+  _FinderBox(this._unboxed);
+
+  final FinderBase<Element> _unboxed;
+
+  @override
+  Iterable<Element> get allCandidates => _unboxed.allCandidates;
+
+  @override
+  Iterable<Element> apply(Iterable<Element> candidates) {
+    return findInCandidates(candidates);
+  }
+
+  @override
+  Finder at(int index) {
+    return _FinderBox(_unboxed.at(index));
+  }
+
+  @override
+  String get description => requirements;
+
+  @override
+  String get requirements => _unboxed.requirements;
+
+  @override
+  Iterable<Element> evaluate() {
+    return find();
+  }
+
+  @override
+  Finder get first => _FinderBox(_unboxed.first);
+
+  @override
+  Finder hitTestable({Alignment at = Alignment.center}) {
+    return _HitTestableFinder(_unboxed, at);
+  }
+
+  @override
+  Finder get last => _FinderBox(_unboxed.last);
+
+  @override
+  bool precache() {
+    if(tryFind()) {
+      return true;
+    } else {
+      _found = null;
+      return false;
+    }
+  }
+
+  @override
+  bool get skipOffstage => !includeOffstage;
+
+  @override
+  bool get includeOffstage {
+    if (_unboxed is ElementFinderMixin) {
+      return (_unboxed as ElementFinderMixin).includeOffstage;
+    } else {
+      return false;
+    }
+  }
+
+  @override
+  FinderResult<Element> find() {
+    return _unboxed.find();
+  }
+
+  @override
+  Iterable<Element> findInCandidates(Iterable<Element> candidates) {
+    return _unboxed.findInCandidates(candidates);
+  }
+
+  @override
+  FinderResult<Element> get found => _unboxed.found;
+
+  @override
+  FinderResult<Element>? _found;
+
+  @override
+  bool get hasFound => _unboxed.hasFound;
+
+  @override
+  bool tryFind() {
+    return _unboxed.tryFind();
+  }
+
+  @override
+  String toString({bool describeSelf = false}) {
+    return _unboxed.toString(describeSelf: describeSelf);
+  }
+}
+
+/// Searches a widget tree and returns nodes that match a particular
+/// pattern.
+// TODO(pdblasi-google): Deprecate Finder. @Deprecated(
+//   'Extend ElementFinder or mixin ElementFinderMixin instead. For variables, type as FinderBase<Element> instead. '
+//   'This will be replaced with the more flexible FinderBase API in a future version of Flutter. '
+//   'This feature was deprecated after v3.11.0-0.0.pre.'
+// )
```

Comment:
Is the deprecation supposed to happen in this PR?

Reply:
I've got a followup ticket (#127136) for actually deprecating things and doing other cleanup. I wanted to keep them a bit separate as the deprecations will get messy and touch a LOT of files.